pkg/agent/multicast/mcast_controller_test.go

```diff
@@ -621,6 +622,115 @@ func TestNodeUpdate(t *testing.T) {
 	}
 }
 
+func TestConcurrentEventHandlerAndWorkers(t *testing.T) {
+	c := newMockMulticastController(t, true)
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	groupIP := net.ParseIP("224.3.4.5")
+	numEvents := 250
+	var wg sync.WaitGroup
+	wg.Add(4)
+	// Below func adds local group join events.
+	go func() {
```

Comment:
Below code is too redundant, can we reduce redundancy?

And there seems no point to generate the events concurrently as they are eventually consumed by a single event handler. The race is not between the event generator and others, but the eventHandler and worker

Reply:
updated.